core/src/test/java/org/infinispan/container/versioning/AbstractClusteredWriteSkewTest.java

```diff
@@ -301,19 +301,22 @@ private void doTestWriteSkewWithPassivation(boolean executeOnPrimaryOwner, Opera
       }
 
       DataContainer dataContainer = TestingUtil.extractComponent(primaryOwner, DataContainer.class);
-      MagicKey[] keys = new MagicKey[MAX_ENTRIES];
+      int retryCount = 5;
       int insertCount = 10;
-      for (int i = 0; i < insertCount; ++i) {
-         for (int j = 0; j < MAX_ENTRIES; ++j) {
-            MagicKey tempKey = keys[j];
-            if (tempKey == null) {
-               tempKey = new MagicKey("other-key-" + j, primaryOwner);
-               keys[j] = tempKey;
+      for (int retry = 0; retry < 5; ++retry) {
```

Comment:
Actually yes there is.  Sadly `insertCount = 50` could still fail.  The biggest difference is that the array is reinitialized causing new keys to be generated https://github.com/infinispan/infinispan/pull/4886/files/1d8d89a0c952c1f7bf2ade50ce18921a08cd4c90#diff-2936358487ff1033e57010e8fff8ca62R307 . This forces caffeine in my test to always evict the desired key after it completes when I tested over 100K times.